Azure/azure-cli
# [ARM] BREAKING CHANGE: `az stack mg create`: Enable mg scoped deployments for mg scoped stacks

**Related command**
az stack mg create

**Description**
MG scoped stacks used to always have the underlying deployment deployed to the default subscription scope (if another subscription is not supplied). We now want the underlying deployment to deploy to the default mg scope if no deployment subscription is provided.

There is also a change to the logic for handling a sub scoped stack with an underlying deployment scoped to a resource group. This was changed to match the first change, as they are the same case of a stack with a lower scoped underlying deployment.

**Testing Guide**
- run az stack mg create -n stack_name -m mg_name--location stack_location --template-file template_file  --deny-settings-mode "none".
- check to ensure deployment is at mg scope.

**History Notes**


[Component Name 1] BREAKING CHANGE: `az stack mg create`: Not supplying '--deployment-scope' will no longer default the underlying deployment to the current subscription scope, but to the mg scope of the deployment stack.
[Component Name 2] `az stack sub create`: Changed internal logic around, but functionally stays the same.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
